x-pack/legacy/plugins/maps/server/plugin.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+import { APP_ID, APP_ICON, createMapPath, MAP_SAVED_OBJECT_TYPE } from '../common/constants';
+import { initRoutes } from './routes';
+import { getEcommerceSavedObjects } from './sample_data/ecommerce_saved_objects';
+import { getFlightsSavedObjects } from './sample_data/flights_saved_objects.js';
+import { getWebLogsSavedObjects } from './sample_data/web_logs_saved_objects.js';
+import { checkLicense } from '../check_license';
+
+export class MapPlugin {
+  setup(core, plugins, __LEGACY) {
+    const { xpackMainPlugin } = __LEGACY.plugins;
+    const { featuresPlugin, licensing } = plugins;
+    let routesInitialized = false;
+
+    featuresPlugin.registerFeature({
+      id: APP_ID,
+      name: i18n.translate('xpack.maps.featureRegistry.mapsFeatureName', {
+        defaultMessage: 'Maps',
+      }),
+      icon: APP_ICON,
+      navLinkId: APP_ID,
+      app: [APP_ID, 'kibana'],
+      catalogue: [APP_ID],
+      privileges: {
+        all: {
+          savedObject: {
+            all: [MAP_SAVED_OBJECT_TYPE, 'query'],
+            read: ['index-pattern']
+          },
+          ui: ['save', 'show', 'saveQuery'],
+        },
+        read: {
+          savedObject: {
+            all: [],
+            read: [MAP_SAVED_OBJECT_TYPE, 'index-pattern', 'query']
+          },
+          ui: ['show'],
+        },
+      }
+    });
+
+    licensing.license$.subscribe(license => {
+      const mapsFeature = license.getFeature('spatial');
+      if (!routesInitialized && mapsFeature.isEnabled) {
+        routesInitialized = true;
+        initRoutes(__LEGACY, license.uid);
+      }
+    });
+
+    xpackMainPlugin.info
+      .feature(APP_ID)
+      .registerLicenseCheckResultsGenerator(checkLicense);
+
+    const sampleDataLinkLabel = i18n.translate('xpack.maps.sampleDataLinkLabel', {
+      defaultMessage: 'Map'
+    });
+    __LEGACY.addSavedObjectsToSampleDataset('ecommerce', getEcommerceSavedObjects());
```

Comment:
I'll update these method implementations as part of the sample data migration to the NP home plugin.